Dockerfile

```diff
@@ -21,19 +33,21 @@ COPY config-at-build-time.json config/config.json
 COPY disclaimer-css.patch /tmp/
 RUN git apply /tmp/disclaimer-css.patch
 
-# `npm install` by itself, with no widget, is necessary before any building
-# because the base image calls `npm prune --production`.
-# `npm install` custom widgets here according to
-# https://enketo.github.io/enketo-express/tutorial-34-custom-widgets.html.
 # Please note that widgets must also be listed in the run-time config.json to
 # be enabled.
-RUN npm install && \
-    npm install https://github.com/kobotoolbox/enketo-image-customization-widget.git && \
-    npm install https://github.com/kobotoolbox/enketo-literacy-test-widget.git && \
-    npx grunt && \
-    npm prune --production && \
-    rm config/config.json
+RUN yarn install --frozen-lockfile \
+    && yarn cache clean \
+    && yarn workspace enketo-express add https://github.com/kobotoolbox/enketo-image-customization-widget#15a1096e2e924cba81caedd69aa17c5920823e26 -W \
+    && yarn workspace enketo-express add https://github.com/kobotoolbox/enketo-literacy-test-widget#8ee65b7ff74f6c2502c86ecd0a1bb8ab3f9670a6 -W \
+    && yarn workspace enketo-express run build \
+    && rm config/config.json
```

Comment:
I'm getting this complaint trying to build locally:
```
error enketo@: The engine "yarn" is incompatible with this module. Expected version ">= 1.22.22 <2". Got "1.22.19"
```

Also, the `cache clean` step should come at the end, after we've done our last `yarn` command. It's job is to avoid making the final Docker image larger than it has to be, but if we clean the cache and continue to use yarn, we'll refill the cache again